sdk/src/azure/platform/az_mosquitto.c

```diff
@@ -229,22 +229,44 @@ AZ_NODISCARD az_result az_mqtt_outbound_connect(az_mqtt* mqtt, az_mqtt_connect_d
   // certs.
   if (az_span_size(me->_internal.options.certificate_authority_trusted_roots) != 0)
   {
-    _az_RETURN_IF_FAILED(_az_result_from_mosq(mosquitto_tls_set(
-        me->_internal.mosquitto_handle,
-        (const char*)az_span_ptr(me->_internal.options.certificate_authority_trusted_roots),
-        NULL,
-        (const char*)az_span_ptr(connect_data->certificate.cert),
-        (const char*)az_span_ptr(connect_data->certificate.key),
-        NULL)));
+    if (connect_data->use_client_certificate)
```

Comment:
Why do we need a separate boolean for this instead of just checking if they've provided connect_data->certificate.cert/connect_data->certificate.key?

I'd also vote to combine these three `mosquitto_tls_set` calls into one since all of the options are something vs NULL

for example, I do this, where ca_file, cert_file, and key_file are NULL if there isn't a value:
```
    if (use_OS_certs)
    {
      MQTT_RETURN_IF_FAILED(mosquitto_int_option(mosq, MOSQ_OPT_TLS_USE_OS_CERTS, true));
    }
    MQTT_RETURN_IF_FAILED(mosquitto_tls_set(
        mosq,
        connection_settings.ca_file,
        use_OS_certs ? REQUIRED_TLS_SET_CERT_PATH : NULL,
        connection_settings.cert_file,
        connection_settings.key_file,
        NULL));
```

Reply:
Removed `use_client_certificate` and added a similar approach to the above.